### ISSUE TITLE
Single-docker container deploy

### ISSUE THREAD

**Issue by amfarrell** [memex-explorer/memex-explorer#609]
We need to have a container made for the entire application, including elasticsearch, tika, kibana, and other containers. This means we will need to have nested containers. See https://blog.docker.com/2013/09/docker-can-now-run-within-docker/.

Plan:
[ ] Run a minimal container within another container, first on EC2 just to prove this out.
[ ] Do that on OSx
[ ] Run the memex-explorer application (without salt states but omitting external containers) in a container.
[ ] Run the whole application, including subordinate containers, within a container.

**Comment by amfarrell** [comment]
@kriehl What is the deadline for when DARPA wants a single-container deploy?

**Comment by ahmadia** [comment]
This is due at the end of the month.

**Comment by ahmadia** [comment]
Needs: https://github.com/memex-explorer/memex-explorer/issues/550

Also needs a video for each "mini-demonstration" of functionality.

**Comment by brittainhard** [comment]
@ahmadia did you succeed with this?

**Comment by bovard** [comment]
I tried to build the docker container locally and looks like the docker file is referencing a branch that I'm not seeing: 
https://github.com/memex-explorer/memex-explorer/blob/master/docker/Dockerfile#L36
and
https://github.com/memex-explorer/memex-explorer/blob/master/docker/Dockerfile#L126

**Comment by ahmadia** [comment]
Thanks @bovard - You should be able to use the master branch to build the Docker container, but you might run into some other problems.  Is there a reason the local development environment (instructions on the front page) won't work for you?

**Comment by ahmadia** [comment]
Specifically I'm referring to the `Local Development` section here: https://github.com/memex-explorer/memex-explorer

**Comment by bovard** [comment]
@ahmadia I'm running into a few issues trying to build this locally and was hoping I could just run the docker :) Do you know if there is a semi-recent docker image that is pushed somewhere that I could pull and run?

**Comment by ahmadia** [comment]
@bovard - I'm much more interested in helping you resolve your local development issues.  We pushed a functional Docker container a month or so ago (see the latest tag here: https://hub.docker.com/r/continuumio/memex-explorer/tags/ ), but the local development is much more up to date.

**Comment by bovard** [comment]
I'll give it another shot locally and let you know what error I run into. Thanks!

**Comment by ahmadia** [comment]
Absolutely, I'm glad you're digging into the code, please don't hesitate to raise issues here if you have any questions or send me an email (You can find it from my homepage)

**Comment by bovard** [comment]
I'll start a new issue for the failure I'm seeing

**Comment by ahmadia** [comment]
I'm updating our Docker image today.
